sonar-orchestrator/src/main/java/com/sonar/orchestrator/build/MavenBuildExecutor.java

```diff
@@ -51,13 +51,18 @@ BuildResult execute(MavenBuild build, Configuration config, Map<String, String>
   private void executeGoal(MavenBuild build, Configuration config, Map<String, String> adjustedProperties, String goal,
     final BuildResult result, CommandExecutor commandExecutor) {
     try {
-      Command command = Command.create(getMvnPath(config.fileSystem().mavenHome()));
+      File mavenHome = config.fileSystem().mavenHome();
+      Command command = Command.create(getMvnPath(mavenHome));
       if (build.getExecutionDir() != null) {
         command.setDirectory(build.getExecutionDir());
       }
       for (Map.Entry<String, String> env : build.getEffectiveEnvironmentVariables().entrySet()) {
         command.setEnvironmentVariable(env.getKey(), env.getValue());
       }
+      if (mavenHome != null) {
+        // Force M2_HOME to override default value from calling env
+        command.setEnvironmentVariable("M2_HOME", mavenHome.getAbsolutePath());
```

Comment:
curious about this: why is m2_home needed if we call mvn with the full path? Is it picked up by sonar mvn plugin?